src/IO/HTTPCommon.cpp

```diff
@@ -33,14 +33,20 @@ namespace ErrorCodes
     extern const int RECEIVED_ERROR_TOO_MANY_REQUESTS;
 }
 
-void setResponseDefaultHeaders(HTTPServerResponse & response, size_t keep_alive_timeout)
+void setResponseDefaultHeaders(HTTPServerResponse & response)
 {
     if (!response.getKeepAlive())
         return;
 
-    Poco::Timespan timeout(keep_alive_timeout, 0);
-    if (timeout.totalSeconds())
-        response.set("Keep-Alive", "timeout=" + std::to_string(timeout.totalSeconds()));
+    const size_t keep_alive_timeout = response.getSession().getKeepAliveTimeout();
```

Comment:
Here we get ka timeout from server's config. I do not see where we are reading ka timeout from client's request.

Reply:
afaiu it was addressed in your PR